include/seqan3/io/structure_file/output.hpp

```diff
@@ -0,0 +1,967 @@
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::BPP)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::STRUCTURE)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::STRUCTURED_SEQ)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::ENERGY)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::REACT)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::REACT_ERR)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::COMMENT)>(t)),
+           detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::OFFSET)>(t)));
+    }
+    //!\}
+
+    //!\brief The options are public and its members can be set directly.
+    structure_file_output_options options;
+
+    /*!\cond DEV
+     * \brief Expose a reference to the underlying stream object. [public, but not documented as part of the API]
+     */
+    stream_type & get_stream()
+    {
+        return stream;
+    }
+    //!\endcond
+protected:
+    //!\privatesection
+    //!\brief Path of the file that the stream operates on.
+    std::string file_name;
+
+    //!\brief The stream we are writing to.
+    stream_type stream;
+
+    //!\brief Type of the format, an std::variant over the `valid_formats`.
+    using format_type = detail::transfer_template_args_onto_t<valid_formats, std::variant>;
+    //!\brief The actual std::variant holding a pointer to the detected/selected format.
+    format_type format;
+
+    //!\brief Write record to format.
+    template <typename seq_type,
+              typename id_type,
+              typename bpp_type,
+              typename structure_type,
+              typename structured_seq_type,
+              typename energy_type,
+              typename react_type,
+              typename comment_type,
+              typename offset_type>
+    void write_record(seq_type && seq,
+                      id_type && id,
+                      bpp_type && bpp,
+                      structure_type && structure,
+                      structured_seq_type && structured_seq,
+                      energy_type && energy,
+                      react_type && react,
+                      react_type && react_error,
+                      comment_type && comment,
+                      offset_type && offset)
+    {
+        static_assert(detail::decays_to_ignore_v<structured_seq_type> ||
+                      (detail::decays_to_ignore_v<seq_type> && detail::decays_to_ignore_v<structure_type>),
+                      "You may not select field::STRUCTURED_SEQ and either of field::SEQ and field::STRUCTURE "
+                      "at the same time.");
```

Comment:
I think the condition is wrong, or not?

Reply:
I cannot find anything wrong here... Either structured_seq must be ignored or both seq and structure.

Reply:
Yep, my bad.